src/main/java/com/kitaab/hisaab/ledger/config/CustomMongoDBConnectionConfigurer.java

```diff
@@ -0,0 +1,98 @@
+package com.kitaab.hisaab.ledger.config;
+
+import com.oracle.bmc.ConfigFileReader;
+import com.oracle.bmc.Region;
+import com.oracle.bmc.auth.AuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimplePrivateKeySupplier;
+import com.oracle.bmc.secrets.SecretsClient;
+import com.oracle.bmc.secrets.model.Base64SecretBundleContentDetails;
+import com.oracle.bmc.secrets.requests.GetSecretBundleRequest;
+import com.oracle.bmc.secrets.responses.GetSecretBundleResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.binary.Base64;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.data.mongo.MongoDataAutoConfiguration;
+import org.springframework.boot.autoconfigure.mongo.MongoAutoConfiguration;
+import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.Supplier;
+
+/**
+ * Configuration to fetch the Mongo Connection url from vault
+ *
+ */
+@Configuration
+@Slf4j
+@AutoConfigureBefore({MongoAutoConfiguration.class, MongoDataAutoConfiguration.class})
+public class CustomMongoDBConnectionConfigurer {
+
+    @Value("${config.credentialConfigFilePath}")
+    private String configFilePath;
+
+    @Bean
+    @Primary
+    public MongoProperties mongoProperties() {
+        var mongoProperties = new MongoProperties();
+        mongoProperties.setUri(getMongoDBConnectionUri());
+        return mongoProperties;
+    }
+
+    public String getMongoDBConnectionUri() {
+        try {
+            ConfigFileReader.ConfigFile config = getConfigFile();
+            Supplier<InputStream> privateKeySupplier
+                    = new SimplePrivateKeySupplier(config.get("key_file"));
+            AuthenticationDetailsProvider provider = getAuthenticationDetailsProvider(config, privateKeySupplier);
```

Comment:
Use ConfigFileAuthenticationDetailsProvider instead of creating a seperate method
Example:
 `AuthenticationDetailsProvider provider = new ConfigFileAuthenticationDetailsProvider(configFile);`